src/Runner.Common/JobServerQueue.cs

```diff
@@ -111,8 +113,11 @@ public void Start(Pipelines.AgentJobRequestMessage jobRequest)
                 !string.IsNullOrEmpty(resultsReceiverEndpoint))
             {
                 Trace.Info("Initializing results client");
-                _jobServer.InitializeResultsClient(new Uri(resultsReceiverEndpoint), accessToken);
-                _resultsClientInitiated = true;
+                if (_resultsServer != null)
```

Comment:
`_resultsServer ` should never be NULL since it's created by `public override void Initialize(IHostContext hostContext)`

Reply:
Removed null guard.